molgenis-core-ui/src/main/java/org/molgenis/ui/metadataeditor/MetadataEditorController.java

```diff
@@ -0,0 +1,144 @@
+import org.molgenis.ui.metadataeditor.model.EditorAttribute;
+import org.molgenis.ui.metadataeditor.model.EditorAttributeResponse;
+import org.molgenis.ui.metadataeditor.model.EditorEntityType;
+import org.molgenis.ui.metadataeditor.model.EditorEntityTypeResponse;
+import org.molgenis.util.ErrorMessageResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.data.i18n.LanguageService.getLanguageCodes;
+import static org.molgenis.ui.metadataeditor.MetadataEditorController.URI;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+
+@Controller
+@RequestMapping(URI)
+public class MetadataEditorController extends MolgenisPluginController
+{
+	private static final Logger LOG = LoggerFactory.getLogger(MetadataEditorController.class);
+
+	public static final String ID = "metadataeditor";
+	public static final String URI = MolgenisPluginController.PLUGIN_URI_PREFIX + ID;
+
+	private final MetaDataService metadataService;
+	private final EntityTypeMapper entityTypeMapper;
+	private final AttributeMapper attributeMapper;
+
+	@Autowired
+	public MetadataEditorController(MetaDataService metadataService, EntityTypeMapper entityTypeMapper,
+			AttributeMapper attributeMapper)
+	{
+		super(URI);
+		this.metadataService = requireNonNull(metadataService);
+		this.entityTypeMapper = requireNonNull(entityTypeMapper);
+		this.attributeMapper = requireNonNull(attributeMapper);
+	}
+
+	@RequestMapping(method = GET)
+	public String init()
+	{
+		return "view-metadataeditor";
+	}
+
+	@RequestMapping(value = "/entityType/{id:.+}", method = GET)
+	@ResponseBody
+	public EditorEntityTypeResponse getEntityType(@PathVariable("id") String entityTypeId)
+	{
+		// metadataService.getEntityType cannot be used due to https://github.com/molgenis/molgenis/issues/5783
+		EntityType entityType = metadataService
+				.getRepository(EntityTypeMetadata.ENTITY_TYPE_META_DATA, EntityType.class).findOneById(entityTypeId);
+
+		return createEntityTypeResponse(entityType);
+	}
+
+	// FIXME currently throws an exception until entity type has an automatically generated identifier
+	@RequestMapping(value = "/create/entityType", method = GET)
+	@ResponseBody
+	public EditorEntityTypeResponse createEntityType()
+	{
+		return createEntityTypeResponse();
+	}
+
+	@RequestMapping(value = "/entityType", method = POST)
+	@ResponseStatus(OK)
+	public void upsertEntityType(@RequestBody EditorEntityType editorEntityType)
+	{
+		EntityType entityType = entityTypeMapper.toEntityType(editorEntityType);
```

Comment:
The EditorEntityType has a different structure and contains different data such that the frontend can be rendered without having to retrieve parts of the data using multiple REST API requests and piecing them together in the client.